### PR TITLE
Add check for template disk size

### LINKED ISSUE #1030: Fail to shrink disk size while cloning VM template
/kind bug

**What steps did you take and what happened:**

1) Clone VM from a template, and specify a lower `DiskGiB` then it's specified on the template
2) Cloning fails with an undescriptive error message: `Invalid operation for device '0'.`

**What did you expect to happen:**

Show an error on specifying an invalid user input, and do not attempt to clone the template.

**Anything else you would like to add:**

Full (almost) error output:
```json
   "Description": {
    "Key": "com.vmware.vim.vpxd.vpx.vmprov.ReconfigureVm",
    "Arg": null,
    "Message": "Reconfiguring Virtual Machine on destination host"
   },
   "Name": "CloneVM_Task",
   "DescriptionId": "VirtualMachine.clone",
   "Entity": {
    "Type": "VirtualMachine",
    "Value": "vm-1880"
   },
   "EntityName": "1909-template",
   "Locked": null,
   "State": "error",
   "Cancelled": false,
   "Cancelable": false,
   "Error": {
    "Fault": {
     "FaultCause": null,
     "FaultMessage": null,
     "Property": "virtualDeviceSpec.device",
     "DeviceIndex": 0,
     "BadOp": "edit",
     "BadFileOp": ""
    },
    "LocalizedMessage": "Invalid operation for device '0'."
   },
```
**Environment:**

- Cluster-api-provider-vsphere version: 
- Kubernetes version: (use `kubectl version`): 1.19
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
**What this PR does / why we need it**:

Prevent users from hitting an undescriptive error message, while attempting to clone a VM from a template with a required disk size less than the disk space specified on the template.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

Fixes #1030 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/services/govmomi/vcenter/clone_test.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2019 The Kubernetes Authors.
```
Comment (gab-satchi) [inline pkg/services/govmomi/vcenter/clone_test.go:2]: should be `Copyright 2020`

**Review by gab-satchi — COMMENTED** [review]
lgtm. just nit on Copyright

File: pkg/services/govmomi/vcenter/clone_test.go (hunk 1 above)
Comment (Danil-Grigorev) [reply pkg/services/govmomi/vcenter/clone_test.go]: Right, should be fixed now, thanks :+1:   PTAL @gab-satchi @randomvariable

**Review by vincepri — APPROVED** [review]
/approve

**Pull request merged**